### DISCUSSION TITLE
numerical simulation shock induced cavitation

### DISCUSSION THREAD

**Discussion by waynezw0618** [idaholab/moose#20700] [Q&A Modules: Phase field] [answered]
Hello Everyone
I am new to moose. I am used to OpenFOAM for hydrodynamics cavitation simulation. I am wondering whether I can use MOOSE to simulate the rarefaction behind shock wave induced cavitation. I suppose the phase filed method is required. some pressure–volumetric strain relation can be adopted for the phase field model.  to capture the shock, I need compressible navier stokes. the viscosity could be neglected.  is there any suggestion ?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You would be welcome to develop this capability in MOOSE. Either:
- off of the compressible finite volume formulations (Kurdanov-Tadmor or HLLC) in the Navier Stokes module. We have not built the link to phase field for this either, I think you would want to re-implement phase field with finite volume OR increase our capabilities for finite volume/finite element coupling. 
- by adding a fully compressible scheme in finite element in the Navier Stokes module. Then you could could couple to phase field more easily.

Guillaume
